sdk/python/kfp/dsl/for_loop.py

```diff
@@ -160,11 +179,16 @@ def from_pipeline_channel(
     ) -> 'LoopArgument':
         """Creates a LoopArgument object from a PipelineParameterChannel
         object."""
+        if channel.channel_type.startswith('typing.Dict'):
+            loop_argument_channel_type = channel.channel_type
+        else:
+            loop_argument_channel_type = _get_loop_item_type(
+                channel.channel_type)
         return LoopArgument(
             items=channel,
             name_override=channel.name + '-' + cls.LOOP_ITEM_NAME_BASE,
             task_name=channel.task_name,
-            channel_type=_get_loop_item_type(channel.channel_type) or 'String',
+            channel_type=loop_argument_channel_type or 'String',
```

Comment:
Can you help me understand for which user code cases we will hit the `or 'String'` fallback?

This commented may be related to the other comment about unit testing the different cases we might encounter.